crates/cli-support/src/js/binding.rs

```diff
@@ -920,6 +933,39 @@ fn instruction(
             js.push(format!("isLikeNone({0}) ? {1} : {0}", val, hole));
         }
 
+        Instruction::F64FromOptionSentinelInt { signed } => {
+            let val = js.pop();
+            js.cx.expose_is_like_none();
+            js.assert_optional_number(&val);
+
+            // We need to convert the given number to a 32-bit integer before
+            // passing it to the ABI for 2 reasons:
+            // 1. Rust's behavior for `value_f64 as i32/u32` is different from
+            //    the WebAssembly behavior for values outside the 32-bit range.
+            //    We could implement this behavior is Rust too, but it's easier
+            //    to do it in JS.
+            // 2. If we allowed values outside the 32-bit range, the sentinel
+            //    value itself would be allowed. This would make it impossible
+            //    to distinguish between the sentinel value and a valid value.
+
+            let op = if *signed { ">>" } else { ">>>" };
+            js.push(format!("isLikeNone({val}) ? 0x100000001 : ({val}) {op} 0"));
```

Comment:
It wasn't immediately clear to me whats going on here.
You might want to mention that bitshift operators in JS always return a 32-bit number.

(I also didn't know about the `>>>` operator, but thats discernible from the code and can easily be looked up)